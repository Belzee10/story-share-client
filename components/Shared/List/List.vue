<template>
  <v-row>
    <v-col v-for="item in items" :key="item.id" :cols="cols">
      <slot :itemProps="item"></slot>
    </v-col>
  </v-row>
</template>

<script>
export default {
  name: 'List',
  props: {
    items: {
      type: Array,
      default: () => []
    },
    cols: {
      type: Number,
      default: null
    }
  }
};
</script>

<style lang="scss" scoped></style>
