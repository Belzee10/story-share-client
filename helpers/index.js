/**
 * transform the response from express
 * remove values from exclude
 * change the key "_id" by "id"
 * @param {Object} item
 * @param {Array} exclude
 */
export const transformResponse = (item, exclude = ['__v', 'created_at']) => {
  Object.keys(item).forEach(key => {
    if (exclude.includes(key)) delete item[key];
  });
  if (item._id) {
    item.id = item._id;
    delete item._id;
  }
  return item;
};
