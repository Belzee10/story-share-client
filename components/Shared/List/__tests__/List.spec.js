import { createWrapper } from '~/config/tests.helpers';
import List from '../List';
import provokeArray from 'provoke-array';

describe('List.vue', () => {
  test('should render correctly', () => {
    const items = provokeArray(3, i => ({ id: i }));
    const wrapper = createWrapper(List, {
      props: {
        items
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
