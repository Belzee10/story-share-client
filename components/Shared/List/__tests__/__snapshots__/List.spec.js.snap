// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue should render correctly 1`] = `
<div
  class="row"
/>
`;
